MozziGuts.cpp

```diff
@@ -156,7 +172,11 @@ inline void advanceADCStep() {
   }
   adc_count++;
 }
-#else
+#endif // #if (AUDIO_INPUT_MODE == AUDIO_INPUT_LEGACY)
+#endif // #if (USE_AUDIO_INPUT == true)
+//#elif (!USE_AUDIO_INPUT || (USE_AUDIO_INPUT && BYPASS_MOZZI_INPUT_BUFFER)) // in case we skipped the previous block, we still need to implement async ADC, just like without audio inputs
```

Comment:
I think this was previously breaking mozziAnalogRead: the else was on `#if (USE_AUDIO_INPUT)`, so if it was true, even if we skipped the setting of the ADCs because of the BYPASS_MOZZI_INPUT_BUFFER all the steps of ADC would not be defined. I think this is correct but it is so easy to mess up this logic that I would appreciate a second look here.